src/network/address.rs

```diff
@@ -110,17 +109,11 @@ impl fmt::Debug for Address {
     }
 }
 
-/// Address received from BIP155 addrv2 message
-#[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub struct AddrV2Message {
-    /// Time that this node was last seen as connected to the network
-    pub time: u32,
-    /// Service bits
-    pub services: ServiceFlags,
-    /// Network ID + Network Address
-    pub addr: AddrV2,
-    /// Network port, 0 if not applicable
-    pub port: u16
+impl ToSocketAddrs for Address {
+    type Iter = option::IntoIter<SocketAddr>;
+    fn to_socket_addrs(&self) -> Result<Self::Iter, io::Error> {
+        self.socket_addr().map(|s| Some(s).into_iter())
```

Comment:
nil, `iter::once(s)` can be easier to understand

Reply:
Hmm, I just checked how the std impls of ToSocketAddrs where implemented and they used `option::IntoIter`.

But I changed it, I agree it's easier to understand.